src/warplib/implementations/conversions/int.ts

```diff
@@ -116,19 +25,24 @@ export function functionaliseIntConversion(conversion: FunctionCall, ast: AST):
     )}`,
   );
 
-  if (fromType.nBits < 256 && toType.nBits === 256 && !fromType.signed && !toType.signed) {
-    IntFunction(conversion, conversion.vArguments[0], 'uint', 'int_conversions', ast);
-    return;
-  } else if (
-    fromType.nBits === toType.nBits ||
-    (fromType.nBits < toType.nBits && !fromType.signed && !toType.signed)
-  ) {
-    arg.typeString = conversion.typeString;
-    ast.replaceNode(conversion, arg);
+  const fromCairoWidth = cairoWidths.find((x) => x >= fromType.nBits)!;
+  const toCairoWidth = cairoWidths.find((x) => x >= toType.nBits)!;
```

Comment:
I understand what you say about the `!` operator. Nonetheless you're assuming that both `cairoWidths` and the `fromType.nBits` will be always correct, something that may not be true.

I think it would be a good idea also to add a failing compilation test for the unsupported `nBits`

Reply:
Genuinely asking, doesn't `nBits` come from `solc-typed-ast`, thus, from the solidity AST. Given that no Solidity code can have wider types that 256 bits, what test would show that?

Reply:
Right now there can be `nBits` that do not exisist in cairoWidth. For example 248, 240, ... . It should throw a `NotSupportedYet` error saying that conversion from x to y is not supported yet.

Reply:
Oh no, it doesn't do that. It gets the least Cairo type able to fit that number. So, for u248 it's u256, for u24 it's u32